### ISSUE TITLE
Files API (LS command) returning incorrect JSON?

### ISSUE THREAD

**Issue by Clay-Ferguson** [ipfs/go-ipfs#7776]
#### Version information:
I'm using docker "ipfs/go-ipfs:release" instance and the current version that shows is this:

```
go-ipfs version: 0.4.23-5b1687d
Repo version: 7
System version: amd64/linux
Golang version: go1.12.16
```
#### Description
I think `/api/v0/files/ls` is giving incorrect response JSON, because it lists my file as type=0, and size=0, when the command below shows that it isn't zero size.

This request proves my file exists:

```
REQUEST:
curl -X POST "http://127.0.0.1:5001/api/v0/files/read?arg=/r/myfolder/node.json"

RESPONSE:
{
  "content" : "test"
}
```

Then this request for the folder it's in appears to return wrong data???

```
REQUEST:
curl -X POST "http://127.0.0.1:5001/api/v0/files/ls?arg=/r/myfolder&long=true"

RESPONSE:
{"Entries":[{"Name":"node.json","Type":0,"Size":0,"Hash":""}]}
```

How can I tell 'node.json' is even a file instead of a folder from this, because
folders also appear to have type=0 and size=0.

EDIT: In case it helps, the file was created with this argument "/api/v0/files/write?arg=/r/myfolder/node.json&create=true&parents=true&truncate=true", so I relied on parents=true to create the two subfolders.

**Comment by aschmahmann** [comment]
@Clay-Ferguson because returning information like "is this object a file or directory" and "what size is this object" require actually accessing the object itself (e.g. could be a network request that involves downloading data) the default is that `ls` only shows you the names of the files/directories. If you pass the `-l` flag you'll get object type and size information.

Side note: The version of go-ipfs you're using is out of date, the latest version is the `latest` tag which currently points at go-ipfs v0.7.0 which is much improved over v0.4.23. Out of curiosity how did you come across the `release` tag name? Wondering if there's some documentation out there that needs updating.

**Comment by Clay-Ferguson** [comment]
I'm not sure I understand, because even though the HTTP API reference makes no mention of an "-l" command, I looked in the command line reference and found that "l" and "long" are the same thing. So I did pass "long=true" on the url (shown above) but it still doesn't populate anything other than "Name" in the response JSON.

I will update my docker image version. I don't remember where I got that "ipfs/go-ipfs:release" one. Maybe this is improved/fixed in v0.7.0. I'll report back what I find. Thanks very much for your time!

**Comment by Clay-Ferguson** [comment]
UPDATE: I ran docker with version v0.7.0 and everything is perfect. It returns the types and sizes. 

Really sorry for wasting my time and yours!! I wondered why it was saying v0.4.23 even with that "release" but I had just assumed docker releases weren't up to date. My mistake.
